tests/LDContextBuilderTest.php

```diff
@@ -0,0 +1,358 @@
+            $context
+        );
+
+        $builder = $builder->name(null);
+
+        $context = $builder->build();
+        $this->assertEquals(
+            new LDContext(
+                LDContext::DEFAULT_KIND,
+                'some-context-key',
+                null,
+                false,
+                null,
+                null,
+                null,
+                null
+            ),
+            $context
+        );
+    }
+
+    public function testAnonymous(): void
+    {
+        $builder = $this->builder->anonymous(true);
+
+        $this->assertEquals($this->builder, $builder);
+
+        $context = $this->builder->build();
+        $this->assertEquals(
+            new LDContext(
+                LDContext::DEFAULT_KIND,
+                'some-context-key',
+                null,
+                true,
+                null,
+                null,
+                null,
+                null
+            ),
+            $context
+        );
+    }
+
+    public function testSet(): void
+    {
+        $builder = $this->builder->set('attribute-name', 'value');
+
+        $this->assertEquals($this->builder, $builder);
+
+        $context = $this->builder->build();
+        $this->assertEquals(
+            new LDContext(
+                LDContext::DEFAULT_KIND,
+                'some-context-key',
+                null,
+                false,
+                ['attribute-name' => 'value'],
+                null,
+                null,
+                null
+            ),
+            $context
+        );
+    }
+
+    /**
+     * @dataProvider trySetDataProvider
+     */
+    public function testTrySet($expectedIsSet, $attributeName, $value, $expectedLdContext): void
+    {
+        $isSet = $this->builder->trySet($attributeName, $value);
+
+        $this->assertEquals($expectedIsSet, $isSet);
+
+        $context = $this->builder->build();
+        $this->assertEquals(
+            $expectedLdContext,
+            $context
+        );
+    }
+
+    public function testPrivate(): void
+    {
+        $builder = $this->builder->private(
+            'string-attribute-ref',
+            AttributeReference::fromPath('refPath'),
+            AttributeReference::fromLiteral('attributeName')
+        );
+        $this->assertEquals($this->builder, $builder);
+
+        $context = $this->builder->build();
+        $this->assertEquals(
+            new LDContext(
+                LDContext::DEFAULT_KIND,
+                'some-context-key',
+                null,
+                false,
+                null,
+                [
+                    AttributeReference::fromPath('string-attribute-ref'),
+                    AttributeReference::fromPath('refPath'),
+                    AttributeReference::fromLiteral('attributeName')
+                ],
+                null,
+                null
+            ),
+            $context
+        );
+    }
+
+    public function testPrivateEmptyAttributeRefs(): void
```

Comment:
We don't have coverage for this exact scenario. I would be okay including this test if you can move it with the others and update to match the testing style in that file.